source/plugins/ruby/kubelet_utils.rb

```diff
@@ -41,6 +41,123 @@ def get_node_capacity
+        capacity_response = CAdvisorMetricsAPIClient.getAllMetricsCAdvisor(winNode: nil)
+        if !capacity_response.nil? && !capacity_response.body.nil?
+          all_metrics = capacity_response.body.split("\n")
+          #cadvisor machine metrics can exist with (>=1.19) or without dimensions (<1.19)
+          #so just checking startswith of metric name would be good enough to pick the metric value from exposition format
+          cpu_capacity = all_metrics.select { |m| m.start_with?("machine_cpu_cores") }.first.split.last.to_f * 1000
+          @log.info "get_node_allocatable::CPU Capacity #{cpu_capacity}"
+          memory_capacity_e = all_metrics.select { |m| m.start_with?("machine_memory_bytes") }.first.split.last
+          memory_capacity = BigDecimal(memory_capacity_e).to_f
+          @log.info "get_node_allocatable::Memory Capacity #{memory_capacity}"
+        end
+
+        allocatable_response = CAdvisorMetricsAPIClient.getCongifzCAdvisor(winNode: nil)
+        parsed_response = JSON.parse(allocatable_response.body)
+
+        begin
+          kubereserved_cpu = parsed_response["kubeletconfig"]["kubeReserved"]["cpu"]
+          if kubereserved_cpu.nil? || kubereserved_cpu == ""
+            kubereserved_cpu = "0"
+          end
+          @log.info "get_node_allocatable::kubereserved_cpu  #{kubereserved_cpu}"
+        rescue => errorStr
+          @log.error "Error in get_node_allocatable::kubereserved_cpu: #{errorStr}"
+          kubereserved_cpu = "0"
+          ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
+        end 
+
+        begin
+          kubereserved_memory = parsed_response["kubeletconfig"]["kubeReserved"]["memory"]
+          if kubereserved_memory.nil? || kubereserved_memory == ""
+            kubereserved_memory = "0"
+          end
+          @log.info "get_node_allocatable::kubereserved_memory #{kubereserved_memory}"
+        rescue => errorStr
+          @log.error "Error in get_node_allocatable::kubereserved_memory: #{errorStr}"
+          kubereserved_memory = "0"
+          ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
+        end 
+        begin
+          systemReserved_cpu = parsed_response["kubeletconfig"]["systemReserved"]["cpu"]
+          if systemReserved_cpu.nil? || systemReserved_cpu == ""
+            systemReserved_cpu = "0"
+          end
+          @log.info "get_node_allocatable::systemReserved_cpu  #{systemReserved_cpu}"
+        rescue => errorStr
+          # this will likely always reach this condition for AKS ~ only applicable for hyrid + MDM combination
+          @log.error "Error in get_node_allocatable::systemReserved_cpu: #{errorStr}"
+          systemReserved_cpu = "0"
+          ApplicationInsightsUtility.sendExceptionTelemetry("Error in get_node_allocatable::kubereserved_cpu: #{errorStr}")
+        end 
+
+        begin
```

Comment:
find better way of doing this ~ right now it logs too much telemetry.

Reply:
actually this is fine, its only happening once when the container starts up and that's exactly the number of times I want to log this.